src/org/mirah/typer/better_closures.mirah

```diff
@@ -477,14 +523,22 @@ class BetterClosureBuilder
 
       
 
-      parent = CallSite(block.parent)
-      replace_block_with_closure_in_call parent, block, new_node
+      if block.parent.kind_of?(CallSite)
+        parent = CallSite(block.parent)
+        replace_block_with_closure_in_call parent, block, new_node
+      else
```

Comment:
I think I'd prefer if this was a elsif checking the type against SyntheticLambdaDefinition, with an else that raises an error.